src/main/java/net/clgd/ccemux/peripherals/PeripheralFactory.java

```diff
@@ -0,0 +1,14 @@
+package net.clgd.ccemux.peripherals;
+
+import dan200.computercraft.api.peripheral.IPeripheral;
+import net.clgd.ccemux.emulation.EmulatedComputer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@FunctionalInterface
+public interface PeripheralFactory<T extends IPeripheral> {
+	public static final Map<String, PeripheralFactory<?>> implementations = new HashMap<>();
```

Comment:
I realise you're following the pattern of `RendererFactory`, but who thought it was okay to expose implementation details like this? We should generally go through all the `*Factory` classes and make a `register` method.

That's probably outside the scope of this PR though, so keep it as it is either way.